presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -553,6 +553,44 @@ public static void multiply(Slice left, Slice right, Slice result)
         setRawInt(result, 7, z7);
     }
 
+    public static Slice multiply(Slice decimal, int multiplier)
+    {
+        Slice result = Slices.copyOf(decimal);
+        multiplyDestructive(result, multiplier);
+        return result;
+    }
+
+    public static void multiplyDestructive(Slice decimal, int multiplier)
+    {
+        int l0 = getInt(decimal, 0);
+        int l1 = getInt(decimal, 1);
+        int l2 = getInt(decimal, 2);
+        int l3 = getInt(decimal, 3);
+
+        int r0 = Math.abs(multiplier);
+
+        long product;
+
+        product = (r0 & LONG_MASK) * (l0 & LONG_MASK);
```

Comment:
please check @sopel39's response above (I'm keeping this code in sync with `multiply` method)